### ISSUE TITLE
Conditional logic w/symbols & execution of bound functions

### ISSUE THREAD

**Issue by Pranab16** [TroyAlford/react-jsx-parser#57]
Can I achieve this ?

    <JsxParser
      bindings={{ a: true }}
      jsx={`
        <div>{a && <span>Print this</span>}</div>
      `}
    />

**Comment by TroyAlford** [comment]
Hmm... probably not at the moment. I think the component would need to handle another type of symbol logic, beyond what it currently supports in order to make this run.

My concern is that once the component starts executing arbitrary JS like this, we open ourselves up to XSS attacks. It's not a far cry to go from this to:
```javascript
<JsxParser jsx={`<div>{true && <span>Print this</span>}</div>`} />
// or
<JsxParser jsx={`<div>{arbitraryFunctionCall() && <span>Print this</span>}</div>`} />
```
And at that point, if someone is able to manipulate the JSX you're saving off to your server-side and then re-rendering, it's pretty simple to start executing their own code. I'd be open to collaborating on a pull request to support what you're after, if we could come up with a clever way to make it safe and limited - but so far, I've staunchly resisted opening the door to this type of code parsing & execution.

**Comment by Pranab16** [comment]
What if we limit this to just binding vars or functions?

**Comment by TroyAlford** [comment]
Hmm... this is probably possible, yes. Acorn can certainly identify a symbol/identifier or a function call. I'm not sure how the `&&` sequence is interpreted - but perhaps the first step is just to write a unit test to exercise what we want to have work, and then walk through the execution with the debugger to figure it out?

**Comment by Pranab16** [comment]
Does it repo require push permission? I am not able to push a branch.

**Comment by TroyAlford** [comment]
Typically what folks do is fork the repo, then make a pull request from the fork back to the main repo

**Comment by Pranab16** [comment]
PR for the same - https://github.com/TroyAlford/react-jsx-parser/pull/58/files

**Comment by TroyAlford** [comment]
Great stuff! Thanks for the contribution. :) I'll get this released shortly.

**Comment by TroyAlford** [comment]
This is now released as of v1.3.3
